digit_main/src/main.cpp

```diff
@@ -328,33 +328,54 @@ void loop() {
         np.update();
     }
 
-    // Wrist Yaw
-    // if (millis() - lastWristYawIter > 175 && (isWristYawIK || wristYawDir != 0)) {
-    //     lastWristYawIter = millis();
-
-    //     if (isWristYawIK && wristYaw != wristYawIKGoal) {  // IK Yaw Control
-    //         // TODO: decimal?
-    //         if (wristYaw < wristYawIKGoal) {
-    //             wristYaw += 1;
-    //             topLSS.moveRelative(-20);
-    //             bottomLSS.moveRelative(20);
-    //         } else if (wristYaw > wristYawIKGoal) {
-    //             wristYaw -= 1;
-    //             topLSS.moveRelative(20);
-    //             bottomLSS.moveRelative(-20);
-    //         }
-    //     } else if (!isWristYawIK && wristYawDir != 0) {  // Manual Yaw Control
-    //         if (wristYawDir == 1) {
-    //             wristYaw += 1;
-    //             topLSS.moveRelative(-20);
-    //             bottomLSS.moveRelative(20);
-    //         } else if (wristYawDir == -1) {
-    //             wristYaw -= 1;
-    //             topLSS.moveRelative(20);
-    //             bottomLSS.moveRelative(-20);
-    //         }
-    //     }
-    // }
+    // Wrist Control
+    if (millis() - lastWristCtrl > 100) {
+        lastWristCtrl = millis();
+
+        float topLSSAngle = topLSS.getPosition() / 10.0;
+        float bottomLSSAngle = bottomLSS.getPosition() / 10.0;
+
+        lastWristYaw = clamp_angle(topLSSAngle - bottomLSSAngle) / 2.0;
+        lastWristRoll = (topLSSAngle + bottomLSSAngle) / (2 * LSS_GEAR_RATIO);
+
+        // IK Control
+        if (isWristCtrlIK) {
+            float k = 1;  // Mechanical constant
+
+            float topTarget = LSS_GEAR_RATIO * wristIKRollGoal + (wristIKYawGoal / (2 * k));
+            float bottomTarget = LSS_GEAR_RATIO * wristIKRollGoal - (wristIKYawGoal / (2 * k));
+
+            topLSS.move(topTarget * 10.0);
+            bottomLSS.move(bottomTarget * 10.0);
+        }
+        // Manual Control
+        else {
+            if (wristManRollDir == 0 && wristManYawDir == 0) {  // Stop as soon as possible if requested
```

Comment:
this would read better as a guard (ie `return` at the end of this `if`). that way the rest of the code isn't in an `else`

Reply:
oh i see

if we extract this functionality out into other functions it could make it more readable, but that might be out of scope for this PR